src/nodes.coffee

```diff
@@ -4525,3 +4659,27 @@ mergeAstLocationData = (nodeA, nodeB) ->
     ]
     start: lesser  nodeA.start, nodeB.start
     end:   greater nodeA.end,   nodeB.end
+
+# Convert Jison-style node class location data to Babel-style location data
+locationDataToAst = ({first_line, first_column, last_line, last_column, range}) ->
+  return
+    loc:
+      start:
+        line:   first_line + 1
+        column: first_column
+      end:
+        line:   last_line + 1
+        column: last_column + 1
+    range: [
+      range[0]
+      range[1]
+    ]
+    start: range[0]
+    end:   range[1]
+
+# Extract location data fields from an AST node
+extractAstLocationData = (ast) ->
+  loc: ast.loc
+  range: ast.range
+  start: ast.start
+  end: ast.end
```

Comment:
This feels like a leftover from the version of the AST branch before we split up into `astLocationData` and other methods.

Reply:
There are still places where it's more convenient to "copy" the location data from an AST node we've already generated than to try and recreate it through calls to `.astLocationData()`